### PR TITLE
Add support for scoped packages in ESLint special. Close #174

### LINKED ISSUE #174: Allow scoped packages for special ESLint
In the special handler for ESLint there is currently an option disabled where it is about scoped packages (see [this line of code](https://github.com/depcheck/depcheck/blob/fc44c61dfcc4e3e64c86bb87ce9e29a7dec8529e/src/special/eslint.js#L47)).

I would like to do something about it, but I am note sure what to do.

Basically, you could just say, that any scoped package is fine here. This would be the easiest change. Do you have any special ideas for this?

Please let me know and I can prepare a PR.

### PR DESCRIPTION
**Pull request by goloroden**
Hi @lijunle 😊 

I have added support for scoped modules. I did not follow the new ESLint code, but rather cleaned up the existing one and added what was needed to make things work with scoped modules.

I hope you like it 😊

### CODE REVIEW

**Comment by goloroden** [comment]
PS: Node.js 0.12 tests pass, but running depcheck on itself doesn't work on this version of Node.js. With 4.x and 6.x, everything is fine. To be honest, I do not actually know why this is :-(

File: src/special/eslint.js [hunk 1]
```diff
@@ -37,19 +37,43 @@ function wrapToArray(obj) {
   return [obj];
 }
 
+function isEslintConfigAnAbsolutePath(specifier) {
+  return path.isAbsolute(specifier);
+}
+
+function isEslintConfigARelativePath(specifier) {
+  // return !/\w|@/.test(preset.charAt(0));
```
Comment (lijunle) [inline src/special/eslint.js]: Could you please clean up this comment?

File: src/special/eslint.js [hunk 2]
```diff
@@ -37,19 +37,43 @@ function wrapToArray(obj) {
...
+function isEslintConfigAScopedModule(specifier) {
+  return specifier.startsWith('@');
+}
+
+function isEslintConfigAFullyQualifiedModuleName(specifier) {
+  return specifier.startsWith('eslint-config-');
+}
+
 function resolvePresetPackage(preset, rootDir) {
-  // inspired from https://github.com/eslint/eslint/blob/5b4a94e26d0ef247fe222dacab5749805d9780dd/lib/config/config-file.js#L347
```
Comment (lijunle) [inline src/special/eslint.js]: Please don't remove this inspired comment. It is a clue for future updates.

File: test/special/eslint.js [hunk 3]
```diff
@@ -96,6 +96,24 @@ const testCases = [
...
+      extends: '@my-org/customized',
+    },
+    expected: [
+      '@my-org/eslint-config-customized',
+    ],
+  },
+  {
+    name: 'handle config of scoped module with full name',
+    content: {
+      extends: '@my-org/eslint-config-customized',
```
Comment (lijunle) [inline test/special/eslint.js]: Could you please switch another random name here? Or code is search friendly.

File: src/special/eslint.js (hunk 1 above)
Comment (goloroden) [inline src/special/eslint.js]: Done.

File: src/special/eslint.js (hunk 2 above)
Comment (goloroden) [inline src/special/eslint.js]: Done.

File: test/special/eslint.js (hunk 3 above)
Comment (goloroden) [inline test/special/eslint.js]: Done.

**Comment by goloroden** [comment]
> The code looks good to me.

Thanks a lot, that's very nice feedback 😊 

> I left the tiny comments. 

I updated all three sections.

> I will look into the failing build.

Thanks!

**Comment by lijunle** [comment]
I get the reason that node 0.12 is failing, see this log: https://travis-ci.org/depcheck/depcheck/jobs/177059242#L614

node 0.12 not suppirt [String.prototype.startsWith](https://github.com/depcheck/depcheck/pull/176/files#diff-b57b843af601dee50a1c35089102d1a3R45).

**Comment by goloroden** [comment]
Awesome, thanks :-)

**Comment by lijunle** [comment]
@goloroden Version 0.6.5 is out, please enjoy it! Thank you again for contributing!

**Comment by goloroden** [comment]
Great news, thanks again 😊

**Pull request merged**